jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile

```diff
@@ -17,15 +17,31 @@ pipeline {
 
         stage('Clone Operator Repo') {
             steps {
-                //ToDo Clone the integeratly-operator repo
-                echo "Clone Operator Repo"
+                cleanWs()
+                git 'https://github.com/integr8ly/integreatly-operator'
             }
         }
 
         stage("Ensure Product Branch") {
             steps {
-                //ToDo Create the product next branch i.e. 3scale-next from the target branch if it doesn't exist
-                echo "Ensure Product Branch"
+                withCredentials([usernamePassword(credentialsId: 'githubjenkins', passwordVariable: 'GIT_PASSWORD', usernameVariable: 'GIT_USERNAME')]) {
+                    sh("""
+                        git config --global user.name 'Automatron'
+                        git config --global user.email 'github.eng@feedhenry.com'
+                        git config --global credential.username ${GIT_USERNAME}
+                        git config --global credential.helper "!echo password=${GIT_PASSWORD}; echo"
+                       """)
+                }
+                script {
+                    def checkout = sh(script: 'git checkout ${PRODUCT_BRANCH}', returnStatus: true)
+                    if (checkout) {
+                        sh("""
+                            git checkout -B ${PRODUCT_BRANCH}
+                            git push origin ${PRODUCT_BRANCH}
```

Comment:
Looks like there's a push in the later stage so maybe it's not needed here?

Reply:
So this is to catch the case where the branch does not exist. I think pushing here is fine because if there's an intermediate step added that reaches out to the remote we'll need the remote to exist.

Reply:
We had this setup originally with DeLorean 1.0 but the number of branches that were hanging around with no changes on them was confusing.

The current approach is the create the branch here if one doesn't already exist, or checkout the existing one, and not to push the branch until the end if changes are detected between it and the base branch (master)